### ISSUE TITLE
Question about cash() 

### ISSUE THREAD

**Issue by quant1729** [pfnet-research/pfhedge#20]
Hello, 

I would like to ask what's the idea behind the cash() of the HedgeLoss ABC? Currently, the Hedger.price() ends up producing the output adjusted for the 'loss' measure, not really the derivative price itself. 

e.g. the following example returns the invalid price of the instrument: 

    torch.manual_seed(42)

    # Prepare a derivative to hedge
    deriv = EuropeanOption(BrownianStock(), strike = 100.0)

    # Create your hedger
    model = MultiLayerPerceptron()
    hedger = Hedger(model, ["log_moneyness", "expiry_time", "volatility", "prev_hedge"])

    # Fit and price
    hedger.fit(deriv, n_paths=10000, n_epochs=200, init_price = 100.0)
    price = hedger.price(deriv, n_paths=10000, init_price = 100.0)
    print(f"Price={price:.5e}")

Can you please elaborate on this idea? I guess I am missing something

**Comment by simaki** [comment]
Thank you for your interest.

The idea behind `HedgeLoss.cash` is the assumption that `HedgeLoss` is a normalized cash equivalent (`\rho` with the notation of the Deep Hedging paper; we use this notation henceforth) and stock price follows the Martingale process.

The output of `HedgeLoss.cash`, denoted `cash`, satisfies `\pi(-Z) = \rho(cash)` provided that the profit-loss is optimized. Then the RHS is shown to be `\rho(cash) = -cash + \rho(0) = -cash + \pi(0)` by the cash-equivalence of `\pi` and Proposition 3.10 in the Deep Hedging paper. A short algebra gives `\pi(-Z - cash) = \pi(0)`, which implies that `-cash` is the derivative price (i.e., the solution to `\pi(-Z + price) = \pi(0)`).

I hope it makes sense.

**Comment by simaki** [comment]
As to the example you provided: `Hedger` by default uses `EntropicRiskMeasure` whose `cash` is overridden by the direct definition of exponential utility indifference price and should be correct.

**Comment by quant1729** [comment]
Sorry for the delay. I think I didn't make my question clear. The output of the code that I have provided is incorrect. It shows the price of 3.7 while it should be around 2.3.

**Comment by simaki** [comment]
Thank you for clarifying.

I think that may be because `MultiLayerPerceptron` has not learned yet to hedge.
In my experience usual feed-forward neural network takes a quite long time to learn.

We can make sure that `Hedger.price()` is correct as follows: In the absence of transaction cost (and at the limit dt -> 0), Black-Scholes' continuous hedging is the optimal hedging strategy. The price corresponding to this strategy is `2.28313e+00` which is around 2.3.

```py
import torch
import pfhedge
from pfhedge import Hedger
from pfhedge.instruments import BrownianStock
from pfhedge.instruments import EuropeanOption
from pfhedge.nn import BlackScholes

torch.manual_seed(42)

deriv = EuropeanOption(BrownianStock(), strike=100.0)

model = BlackScholes(deriv)
hedger = Hedger(model, model.features())

price = hedger.price(deriv, n_paths=10000, init_price=100.0)
print(f"Price={price:.5e}")
```

**Comment by quant1729** [comment]
Thank you for your message. I guess this is a bug because I see that the price is not converging to 2.3 with more epochs/paths.

**Comment by simaki** [comment]
You may want to increase the number of units per layer. Also, `prev_hedge` is not necessary for this case. I got `Price=2.32866e+00`. Sorry that the example in README.md is misleading.

```py
import pfhedge
import torch
from pfhedge import Hedger
from pfhedge.instruments import BrownianStock
from pfhedge.instruments import EuropeanOption
from pfhedge.nn import BlackScholes
from pfhedge.nn import MultiLayerPerceptron

torch.manual_seed(42)

deriv = EuropeanOption(BrownianStock(), strike=100.0)

model = MultiLayerPerceptron(n_units=128)
hedger = Hedger(model, ["log_moneyness", "expiry_time", "volatility"])

hedger.fit(deriv, n_paths=10000, n_epochs=200, init_price=100.0)
price = hedger.price(deriv, n_paths=10000, init_price=100.0)
print(f"Price={price:.5e}")
```
